server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2798,6 +2805,21 @@ long getNumDocUpdates() {
         }
     }
 
+    public int countNumberOfHistoryOperations(String source, long fromSeqNo, long toSeqNo) throws IOException {
+        ensureOpen();
+        refreshIfNeeded(source, toSeqNo);
+        try (Searcher s = acquireSearcher(source, SearcherScope.INTERNAL)) {
+            return LuceneChangesSnapshot.countNumberOfHistoryOperations(s, fromSeqNo, toSeqNo);
+        } catch (Exception e) {
```

Comment:
Can this be scoped down to only IOException?  Do we need to fail the engine on IllegalArgumentException as well?

Reply:
Not a bad idea. I don't think we need to fail on IAE. I used that instead of an assertion since asserts can be disabled so this wide net may be overkill.